commands/update/nodepool/command.go

```diff
@@ -139,17 +138,8 @@ func verifyPreconditions(args Arguments) error {
 	if args.ScalingMin <= 0 && args.ScalingMax <= 0 && args.Name == "" {
 		return microerror.Maskf(errors.NoOpError, "Nothing to update.")
 	}
-
-	switch args.Provider {
-	case provider.AWS:
-		if args.ScalingMin > args.ScalingMax && args.ScalingMax > 0 {
-			return microerror.Mask(errors.WorkersMinMaxInvalidError)
-		}
-
-	case provider.Azure:
-		if args.ScalingMin != args.ScalingMax {
-			return microerror.Maskf(errors.WorkersMinMaxInvalidError, "Provider '%s' does not support node pool autoscaling.", args.Provider)
-		}
+	if args.ScalingMin > args.ScalingMax && args.ScalingMax > 0 {
```

Comment:
why does `args.ScalingMax` need to be >= 0 for this to be an error?

Reply:
I think the reason for doing that is to only throw the error if the `--nodes-max` flag is set (it has a default value of 0).

The command allows the user to set just the min scaling or just the max scaling for a node pool. (so if you want, you can only specify `--nodes-min` or `--nodes-max`). If we didn't check that the `--nodes-max` flag was set, we would throw an error if the user tried to use the `--nodes-min` flag alone.

Hope it makes sense

Reply:
Yes, I understand now.